pkg/common/cns-lib/volume/manager.go

```diff
@@ -557,6 +557,11 @@ func (m *defaultManager) DeleteVolume(ctx context.Context, volumeID string, dele
 		}
 		volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 		if volumeOperationRes.Fault != nil {
+			_, ok := volumeOperationRes.Fault.Fault.(cnstypes.CnsVolumeNotFoundFault)
```

Comment:
Isn't the same thing already handled on line 533? It captures the `vim.fault.NotFound` and returns success.
Or it's different that a task may still be submitted and then we get an error from the task?

Reply:
We already have this check. Isn't this sufficient? Or is this changed due to the backend change?

```go
// Call the CNS DeleteVolume
		cnsVolumeIDList = append(cnsVolumeIDList, cnsVolumeID)
		task, err := m.virtualCenter.CnsClient.DeleteVolume(ctx, cnsVolumeIDList, deleteDisk)
		if err != nil {
			if cnsvsphere.IsNotFoundError(err) {
				log.Infof("VolumeID: %q, not found. Returning success for this operation since the volume is not present", volumeID)
				return nil
			}
			log.Errorf("CNS DeleteVolume failed from the  vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
			return err
}
```

Reply:
Looks like we need to handle error for operationResult after task completes. This is missing in the current implementation